Sources/TensorFlow/Layers/Normalization.swift

```diff
@@ -71,16 +71,26 @@ public struct BatchNorm<Scalar: TensorFlowFloatingPoint>: Layer {
     /// - Returns: The output.
     @differentiable
     public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+        let positiveAxis = (input.rank + axis) % input.rank
+        var offset = self.offset
+        var scale = self.scale
+        if positiveAxis != input.rank - 1 {
+            var broadcastShape = TensorShape([Int](repeating: 1, count: input.rank))
+            broadcastShape[positiveAxis] = input.shape[positiveAxis]
+            offset = offset.reshaped(to: broadcastShape)
+            scale = scale.reshaped(to: broadcastShape)
+            runningMean.value = runningMean.value.reshaped(to: broadcastShape)
+            runningVariance.value = runningVariance.value.reshaped(to: broadcastShape)
```

Comment:
Are these reshapes of `runningMean.value` and `runningVariance.value` necessary?  I think either have rank 0 or the same rank as input, so they already always broadcast correctly.

Reply:
Good point. I removed these reshapes.